tests/manager/repositories/vfolder/test_vfolder_repository.py

```diff
@@ -0,0 +1,324 @@
+        """Create VFolderCreateParams for testing."""
+        return VFolderCreateParams(
+            id=folder_id,
+            name=f"test-model-{folder_id.hex[:8]}",
+            domain_name=domain_name,
+            quota_scope_id=f"project:{group_id}",
+            usage_mode=usage_mode,
+            permission=permission,
+            host="local",
+            creator=f"test-{user_id.hex[:8]}@example.com",
+            ownership_type=VFolderOwnershipType.GROUP,
+            user=user_id,
+            group=group_id,
+            unmanaged_path=None,
+            cloneable=False,
+            status=VFolderOperationStatus.READY,
+        )
+
+    @pytest.fixture
+    def cleanup_tracker(self) -> _CleanupTracker:
+        """Create a cleanup tracker for tracking created entities."""
+        return _CleanupTracker()
+
+    # TODO: Simplify this after removing all fk constraints
+    @pytest.fixture
+    async def db_with_cleanup(
+        self,
+        database_engine: ExtendedAsyncSAEngine,
+        cleanup_tracker: _CleanupTracker,
+    ) -> AsyncGenerator[ExtendedAsyncSAEngine, None]:
+        """
+        Database engine that auto-cleans all test data after each test.
+
+        Due to FK constraints (VFolderRow references UserRow and GroupRow),
+        cleanup must happen in specific order. This fixture cleans up
+        all test-created data in the correct FK order.
+        """
+        yield database_engine
+
+        # Cleanup all test data in correct FK order
+        async with database_engine.begin_session() as db_sess:
+            # 1. VFolder data (references User and Group)
+            await db_sess.execute(sa.delete(VFolderPermissionRow))
+            await db_sess.execute(sa.delete(VFolderRow))
+
+            # 2. Group (referenced by VFolderRow)
+            for group_id in cleanup_tracker.group_ids:
+                await db_sess.execute(sa.delete(GroupRow).where(GroupRow.id == group_id))
+
+            # 3. User (referenced by VFolderRow)
+            for user_id in cleanup_tracker.user_ids:
+                await db_sess.execute(sa.delete(UserRow).where(UserRow.uuid == user_id))
+
+            # 4. Resource policies (referenced by User and Group)
+            for policy_name in cleanup_tracker.project_policy_names:
+                await db_sess.execute(
+                    sa.delete(ProjectResourcePolicyRow).where(
+                        ProjectResourcePolicyRow.name == policy_name
+                    )
+                )
+            for policy_name in cleanup_tracker.user_policy_names:
+                await db_sess.execute(
+                    sa.delete(UserResourcePolicyRow).where(
+                        UserResourcePolicyRow.name == policy_name
+                    )
+                )
+
+            # 5. Domain (referenced by User and Group)
+            for domain_name in cleanup_tracker.domain_names:
+                await db_sess.execute(sa.delete(DomainRow).where(DomainRow.name == domain_name))
```

Comment:
Since it only persists for the fixture lifecycle anyway, let's just delete everything rather than bothering to check before deleting.